extensions/oidc/deployment/src/main/java/io/quarkus/oidc/deployment/OidcBuildStep.java

```diff
@@ -107,6 +118,37 @@ public void provideSecurityInformation(BuildProducer<SecurityInformationBuildIte
                 .produce(SecurityInformationBuildItem.OPENIDCONNECT("quarkus.oidc.auth-server-url"));
     }
 
+    @BuildStep
+    void checkClaim(BeanRegistrationPhaseBuildItem beanRegistrationPhase,
+            BuildProducer<BeanConfiguratorBuildItem> beanConfigurator) {
+
+        for (InjectionPointInfo injectionPoint : beanRegistrationPhase.getContext().get(BuildExtension.Key.INJECTION_POINTS)) {
+            if (injectionPoint.hasDefaultedQualifier()) {
+                continue;
+            }
+            AnnotationInstance claimQualifier = injectionPoint.getRequiredQualifier(CLAIM_NAME);
+            if (claimQualifier != null) {
+                Type actualType = injectionPoint.getRequiredType();
+
+                Optional<BeanInfo> bean = injectionPoint.getTargetBean();
+                if (bean.isPresent()) {
+                    DotName scope = bean.get().getScope().getDotName();
+                    if (!REQUEST_SCOPED_NAME.equals(scope)
```

Comment:
@mkouba I think it works, but better check. Would this be fine with Web Sockets?

Reply:
claim producers are dependent and they require `JsonWebToken` which require request scoped `SecurityIdentity` which is set via `SecuritySupport` everytime that `@OnTextMessage` / binary is invoked. Fact that identity is not changeable between these invocations (in difference CDI request contexts it is still same identity) doesn't change much as it is just an implementation details.

That I read in a way that these claims cannot be injected into session scoped as a type that is not proxiable.

I'll let @mkouba decide as he understands it better.

Reply:
What's the question here? If it's ok to inject `@Dependent` claim bean in a bean that is not `@RequestScoped`? I think it's not  because it's bound to an HTTP request, even though the bean is `@Dependent`.

Reply:
@mkouba @cescoffier thanks, so what is the situation, do you approve this PR or something else is needed to be fixed ? 

The fix is about failing at the build time when primitive type Claims are injected into non `RequestScoped` beans, like `@Inject   @Claim(standard = Claims.upn)  String upn;`.
I'm not sure how the fact WebSockets Next uses SessionScoped beans may be impacting this fix.
This is probably what Michal implied as well...
If it is a String or other primitive type Claim injection into a non `@RequestScoped` bean is done then it is a build time failure...
It is a base level fix which can be tuned further should such a build time failure proves too strict an approach with some other variations